pkg/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -1091,18 +1061,36 @@ func (n *NetworkPolicyController) syncAddressGroup(key string) error {
 		utilsets.MergeString(addrGroupNodeNames, internalNP.SpanMeta.NodeNames)
 	}
 	memberSet := n.getAddressGroupMemberSet(addressGroup)
+	if addressGroup.Selector.NodeSelector != nil {
+		ms, nodes := n.addNodeSelectorMemberSet(addressGroup.Selector.NodeSelector)
+		memberSet = memberSet.Union(ms)
+		utilsets.MergeString(addrGroupNodeNames, nodes)
```

Comment:
I think you misunderstand the meaning of `addrGroupNodeNames`, which is the span of the addressGroup.

Reply:
Updated.